tests/unit/package_managers/test_pip.py

```diff
@@ -3417,15 +3460,22 @@ def test_download_from_requirement_files(
         pypi_download1 = pip_deps.join_within_root("foo", "foo-1.0.0.tar.gz").path
         pypi_download2 = pip_deps.join_within_root("bar", "bar-0.0.1.tar.gz").path
 
-        pypi_info1 = {"package": "foo", "version": "1.0.0", "path": pypi_download1}
-        pypi_info2 = {"package": "bar", "version": "0.0.1", "path": pypi_download2}
+        pypi_package1 = pip.DistributionPackageInfo(
+            "foo", "1.0.0", "sdist", pypi_download1, "", False
+        )
+        pypi_package2 = pip.DistributionPackageInfo(
+            "bar", "0.0.1", "sdist", pypi_download2, "", False
+        )
 
-        mock_pypi_download.side_effect = [pypi_info1, pypi_info2]
+        mock_distributions.side_effect = [(pypi_package1, []), (pypi_package2, [])]
 
         downloads = pip._download_from_requirement_files(rooted_tmp_path, [req_file1, req_file2])
-        assert downloads == [pypi_info1 | {"kind": "pypi"}, pypi_info2 | {"kind": "pypi"}]
+        assert downloads == [
+            pypi_package1.download_info | {"kind": "pypi"},
+            pypi_package2.download_info | {"kind": "pypi"},
+        ]
```

Comment:
Just realized that we do we not return any info about the download wheels. That will be a problem once we stop failing the build when there are 0 sdists. But we can deal with it when we get there

Reply:
I had to rebase and resolve some conflicts....could you just quickly check if I did not break something ?

Reply:
We need to update the data for the new integration test that was added on #291, since pypi_simples does some type of normalization on the package names.